cmd/signing.go

```diff
@@ -0,0 +1,281 @@
+package cmd
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/ipfs/go-log"
+
+	"github.com/keep-network/keep-core/pkg/net"
+	"github.com/keep-network/keep-core/pkg/net/key"
+	"github.com/keep-network/keep-core/pkg/net/local"
+	"github.com/keep-network/keep-ecdsa/pkg/ecdsa"
+	"github.com/keep-network/keep-ecdsa/pkg/ecdsa/tss"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/keep-network/keep-common/pkg/persistence"
+	"github.com/keep-network/keep-ecdsa/internal/config"
+	"github.com/keep-network/keep-ecdsa/pkg/registry"
+	"github.com/urfave/cli"
+)
+
+// SigningCommand contains the definition of the `signing` command-line
+// subcommand and its own subcommands.
+var SigningCommand cli.Command
+
+func init() {
+	SigningCommand = cli.Command{
+		Name:  "signing",
+		Usage: "Provides several tools useful for out-of-band signing",
+		Before: func(c *cli.Context) error {
+			_ = log.SetLogLevel("*", "fatal") // disable the regular logger
+			return nil
+		},
+		Subcommands: []cli.Command{
+			{
+				Name:      "decrypt-key-share",
+				Usage:     "Decrypts the key share of the operator for the given keep",
+				ArgsUsage: "[keep-address]",
+				Action:    DecryptKeyShare,
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:  "output-file,o",
+						Usage: "Output file for the decrypted key share",
+					},
+				},
+			},
+			{
+				Name:      "sign-digest",
+				Usage:     "Sign a given digest using provided key shares",
+				Action:    SignDigest,
+				ArgsUsage: "[unprefixed-hex-digest] [key-shares-dir]",
+			},
+		},
+	}
+}
+
+// DecryptKeyShare decrypt key shares for given keep using provided operator config.
+func DecryptKeyShare(c *cli.Context) error {
+	config, err := config.ReadConfig(c.GlobalString("config"))
+	if err != nil {
+		return fmt.Errorf("failed while reading config file: [%v]", err)
+	}
+
+	keepAddressHex := c.Args().First()
+	if !common.IsHexAddress(keepAddressHex) {
+		return fmt.Errorf("invalid keep address")
+	}
+
+	keepAddress := common.HexToAddress(keepAddressHex)
+
+	handle, err := persistence.NewDiskHandle(config.Storage.DataDir)
+	if err != nil {
+		return fmt.Errorf(
+			"failed while creating a storage disk handler: [%v]",
+			err,
+		)
+	}
+
+	persistence := persistence.NewEncryptedPersistence(
+		handle,
+		config.Ethereum.Account.KeyFilePassword,
+	)
+
+	keepRegistry := registry.NewKeepsRegistry(persistence)
+
+	keepRegistry.LoadExistingKeeps()
```

Comment:
Actually a random note I missed making: we will need to make this work more flexibly. Right now you need a full config file and to have your path set up just right to decrypt the keyshare. We should be able to just point this to a keyshare and, if we have the password configured properly, have that keyshare decrypted to stdout. Otherwise if we have a snapshot that didn't get properly persisted, for example, you'd have to move it into the `current` directory to make this work.

For followup work though.

Reply:
Agree but from the other side, seems we won't be able to use the existing registry code as we do here and we will be forced to copy-paste relevant parts.

Reply:
Yeah I figured structuring it all right will be a bit more complicated.